sql/core/src/test/resources/sql-tests/inputs/misc-functions.sql

```diff
@@ -8,3 +8,12 @@ select typeof(cast(1.0 as float)), typeof(1.0D), typeof(1.2);
 select typeof(date '1986-05-23'),  typeof(timestamp '1986-05-23'), typeof(interval '23 days');
 select typeof(x'ABCD'), typeof('SPARK');
 select typeof(array(1, 2)), typeof(map(1, 2)), typeof(named_struct('a', 1, 'b', 'spark'));
+
+-- Spark-32793: Rewrite AssertTrue with RaiseError
```

Comment:
Could you add tests like this?
```
scala> sql("select * from t").show()
+---+
|  v|
+---+
|  1|
|  8|
|  2|
+---+

scala> sql("select if(v > 5, raise_error('error found: ' || v), v + 1) from t").show()
java.lang.RuntimeException: error found: 8
  at org.apache.spark.sql.catalyst.expressions.RaiseError.eval(misc.scala:80)
```
That is because I think it is a common usecase to print out an error message with an invalid value.